parlai/core/metrics.py

```diff
@@ -231,9 +231,42 @@ def value(self) -> float:
         return self._numer / self._denom
 
 
+class MacroAverageMetric(Metric):
+    """
+    Class that represents the macro average of several numbers.
+
+    Used for aggregating task level metrics.
+    """
+
+    __slots__ = ('_values',)
+
+    def __init__(self, metrics: List[Metric]) -> None:
+        self._values = metrics
+
+    def __add__(self, other: Optional['MacroAverageMetric']) -> 'MacroAverageMetric':
+        if other is None:
+            return self
+        if len(self._values) != len(other._values):
+            raise AssertionError(
+                "MacroAverage keeping track of an uneven number of submetrics. "
+                "There should be exactly one per task."
+            )
+        return MacroAverageMetric([a + b for a, b in zip(self._values, other._values)])
+
+    def value(self) -> float:
+        sum_ = sum(v.value() for v in self._values)
+        n = len(self._values)
+        return sum_ / n
+
+
 class GlobalMetric:
     """
     A global metric is one that should not be aggregated across different tasks.
+
+    Key to it is the notion that any one worker or any one task already has a global
+    view of the value, and so no combinations should be done. Note this is different
+    then a FixedMetric, in that a GlobalMetric can be still averaged across multiple
+    parleys(), but a FixedMetric is always fixed.
```

Comment:
can you give examples of each of these metrics in the descriptiosn? I find it slightly confusing

Reply:
Done for all the metrics classes.